### ISSUE TITLE
Can't die while assisting NPC

### ISSUE THREAD

**Issue by jjhankins** [endless-sky/endless-sky#3206]
```
mission "Test"
	name "Assist the derelict ship Test of Betrayal"
	landing

	npc assist
		personality derelict
		ship Bounder "Test of Betrayal"
		conversation
			`Something about the ship seems odd. Do go on board to assist with repairs?`
			choice
				`	(Yes.)`
				`	(no.)`
					goto end

			`	Pirates spring out of hiding and shoot you dead.`
				die

			label end
			`	You trust your instincts and flee the ship.`

	on offer
		conversation
			`Captain <last>! The ship 'Test of Betrayal' needs your help, please go and help it with its repairs.`
				accept
```

While if you enter the ship you do not see the "You trust your instincts" line, you also don't die. You're dropped back into the system, the derelict ship having been repaired.

EDIT: As a bonus, it would also be nice to destroy the ship and/or merely fail to repair it at the end of the conversation.

**Comment by tehhowch** [comment]
This is because there is no callback function for NPC completion conversations, as mentioned in https://github.com/endless-sky/endless-sky/issues/3129#issuecomment-338011593

PlayerInfo::MissionCallback is what handles `launch`, `flee`, `die`, and other conversation actions:
https://github.com/endless-sky/endless-sky/blob/aad6f581e031e8eb71781180ff5a0e7a175b9e6d/source/PlayerInfo.cpp#L1524-L1558

As I mention in #3129, this is something I'll leave to MZ to address

**Comment by jjhankins** [comment]
Ah, didn't realize that was the same issue.

**Comment by tehhowch** [comment]
It wasn't explicitly stated there.
Even if destruction of that NPC ship from the conversation is not handled, `Die` certainly should be (so keep this open).

**Comment by tehhowch** [comment]
Made possible with #3376

**Comment by endless-sky** [comment]
Addressed by merging #3376.
